### DISCUSSION TITLE
Preserve case of prefix

### DISCUSSION THREAD

**Discussion by ddoherty03** [company-mode/company-mode#1226] [Q&A]
I have the following variables set:
```
  (use-package company
    :ensure t
    :diminish  " ©"
    :custom
    (company-dabbrev-downcase nil)
    (company-dabbrev-ignore-case keep-prefix)
    (company-dabbrev-other-buffers t)
  ...
```
And my company-backends is set to:
```
(company-files company-capf
               (company-yasnippet :with company-dabbrev company-ispell))
```
What I want is for company to complete without down-casing what I've typed so far.  For example:

When I type

Lar

One of the completions is 'larger'.  I want it to complete to 'Larger', but it insists on down-casing my prefix.

What am I doing wrong?

**Comment by dgutov** [comment]
> (company-dabbrev-ignore-case keep-prefix)

Is there a `'` before `keep-prefix`? Otherwise it looks like this line should error out.

> (company-yasnippet :with company-dabbrev company-ispell)

You can also try removing `company-ispell` from this list.

**Reply by ddoherty03** [reply]
@dgutov, I had it with and without the `'`, but have put the quote back in and tried removing `company-ispell` from the backends, but still the same behavior.  This quirk bites me all the time at the beginning of sentences where I have to backup, recapitalize, then move on.

Thanks for taking a look, though.

**Reply by dgutov** [reply]
Looking more into it, I was wrong, sorry. It's the same problem as described in https://github.com/company-mode/company-mode/issues/1096#issuecomment-824949311.

See the comments there for details.

**Reply by ddoherty03** [reply]
Got it.  Thanks, @dgutov.  Any plans to take a run at it?

**Reply by yugaego** [reply]
> Any plans to take a run at it?

It waits for a willing volunteer. Dmitry has provided hints on where to start in the already linked comment and in the following one https://github.com/company-mode/company-mode/issues/1096#issuecomment-824979656.
I personally have this issue on my list, but have got other priorities to work on so far.
